internal/config/config.go

```diff
@@ -27,34 +41,75 @@ type OracleConfig struct {
 // NewConfig ... Initializer
 func NewConfig(fileName FilePath) *Config {
 	if err := godotenv.Load(string(fileName)); err != nil {
-		log.Printf("Config file not found for file name: %s", fileName)
-		panic(err)
+		log.Fatalf("config file not found for file: %s", fileName)
 	}
 
-	return &Config{
-		L1RpcEndpoint: getEnv("L1_RPC_ENDPOINT"),
-		L2RpcEndpoint: getEnv("L2_RPC_ENDPOINT"),
+	config := &Config{
+		L1RpcEndpoint: getEnvStr("L1_RPC_ENDPOINT"),
+		L2RpcEndpoint: getEnvStr("L2_RPC_ENDPOINT"),
+
+		Environment: Env(getEnvStr("ENV")),
+	}
+
+	config.LoggerConfig = &logger.Config{
+		UseCustom:         getEnvBool("LOGGER_USE_CUSTOM"),
+		Level:             getEnvInt("LOGGER_LEVEL"),
+		IsProduction:      config.Environment == Production,
+		DisableCaller:     getEnvBool("LOGGER_DISABLE_CALLER"),
+		DisableStacktrace: getEnvBool("LOGGER_DISABLE_STACKTRACE"),
+		Encoding:          getEnvStr("LOGGER_ENCODING"),
+		OutputPaths:       getEnvSlice("LOGGER_OUTPUT_PATHS"),
+		ErrorOutputPaths:  getEnvSlice("LOGGER_ERROR_OUTPUT_PATHS"),
+
+		EncoderTimeKey:          getEnvStr("LOGGER_ENCODER_TIME_KEY"),
+		EncoderLevelKey:         getEnvStr("LOGGER_ENCODER_LEVEL_KEY"),
+		EncoderNameKey:          getEnvStr("LOGGER_ENCODER_NAME_KEY"),
+		EncoderCallerKey:        getEnvStr("LOGGER_ENCODER_CALLER_KEY"),
+		EncoderFunctionKey:      getEnvStr("LOGGER_ENCODER_FUNCTION_KEY"),
+		EncoderMessageKey:       getEnvStr("LOGGER_ENCODER_MESSAGE_KEY"),
+		EncoderStacktraceKey:    getEnvStr("LOGGER_ENCODER_STACKTRACE_KEY"),
+		EncoderSkipLineEnding:   getEnvBool("LOGGER_ENCODER_SKIP_LINE_ENDING"),
+		EncoderLineEnding:       getEnvStr("LOGGER_ENCODER_LINE_ENDING"),
+		EncoderConsoleSeparator: getEnvStr("LOGGER_ENCODER_CONSOLE_SEPARATOR"),
 	}
+
+	return config
 }
 
-// getEnv ... Reads env var from process environment, panics if not found
-func getEnv(name string) string {
-	envVar := os.Getenv(name)
+// getEnvStr ... Reads env var from process environment, panics if not found
+func getEnvStr(key string) string {
+	envVar, ok := os.LookupEnv(key)
+
 	// Not found
-	if envVar == "" {
-		panic(fmt.Sprintf("Could not find env var for %s", name))
+	if !ok {
+		log.Fatalf("could not find env var given key: %s", key)
 	}
 
 	return envVar
 }
 
-// func convertToInt(str string) int {
-// 	intRep, err := strconv.Atoi(str)
-
-// 	if err != nil {
-// 		panic(err)
-// 	}
+// getEnvBool .. Reads env vars and converts to booleans, panics if incorrect input
+func getEnvBool(key string) bool {
+	if val := getEnvStr(key); val == "1" {
+		return true
+	} else if val == "0" {
+		return false
+	} else {
+		log.Fatalf("env val is not boolean (0 or 1); got: %s=%s", key, val)
```

Comment:
Idk if panicing when a boolean variable isn't 0 or 1 is a correct way to do interpretation here. Typically, you could express `false` by ignoring the env var in your config which this existing logic doesn't currently compensate for. 

Any thoughts? @nadir-akhtar-coinbase